packages/web-config-server/src/models/helpers/EntityHierarchyBuilder.js

```diff
@@ -2,7 +2,9 @@
  * Tupaia
  * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
  */
+import { AsyncTaskQueue } from '@tupaia/utils';
 
+const BATCH_SIZE = 20; // tuned to have optimal performance
```

Comment:
This is the result of some testing (all times in ms):

batch size 1:
Descendants fetch: 34053
Ancestors fetch: 243628

batch size 5:
Descendants fetch: 39514
Ancestors fetch: 90035

batch size 10:
Descendants fetch: 26216
Ancestors fetch: 72655

batch size 20:
Descendants fetch: 26948
Ancestors fetch: 63687

batch size 50:
Descendants fetch: 35432
Ancestors fetch: 81179

batch size 10000:
Descendants fetch: 33900
Ancestors fetch: 77408